src/savefile.c

```diff
@@ -82,19 +82,7 @@ char *extract_dumpname_from_save_file(char *lcname, gboolean *modified, int *opt
 char *append_array_row(GwNode *n)
 {
     char *hname = n->nname;
-
-#ifdef WAVE_ARRAY_SUPPORT
-    if (!n->array_height)
-#endif
-    {
-        strcpy(GLOBALS->buf_menu_c_1, hname);
-    }
-#ifdef WAVE_ARRAY_SUPPORT
-    else {
-        sprintf(GLOBALS->buf_menu_c_1, "%s{%d}", hname, n->this_row);
-    }
-#endif
-
+    strcpy(GLOBALS->buf_menu_c_1, hname);
```

Comment:
The same as `AddNodeUnroll` also applies here. But the required changes might be a bit more complex than just replacing `append_array_row(n)` with `n->nname` everywhere, because we need to ensure that the code that uses the returned string doesn't modify it. I have been caught out by issues where a string has been unexpectedly modified in the past and the compiler didn't catch the issue because the GTKWave codebase doesn't use the `const` keyword to prevent this.

Reply:
Great, I am working on this.